threadpoolctl.py

```diff
@@ -205,10 +206,11 @@ def __exit__(self, type, value, traceback):
         self.unregister()
 
     def unregister(self):
```

Comment:
Beyond the naming of this method, I think most of (all?) the methods of the `_threadpool_limits` class should be marked private. Just in case people start with write code such as:

```python
with threadpool_limits(1) as ctx:
    ctx.unregister()
```

Reply:
If all methods are private, why not simply return None in `__enter__`? this way it is not possible to use the `as ctx`?

Reply:
as discussed irl with @ogrisel, they are already part of the public API so we'll keep them public. In addition `unregister` is useful to restore the limits when it was used as a callable.

It's useful to have access to the object inside the context, at least for testing purpose. I'd rather keep that.

Reply:
Alright. Let's keep the existing method public with an alias with a better name for unregister but no need to deprecate the old name.

Reply:
Ok!